### PR TITLE
Add a test to exhibit #326.

### PR DESCRIPTION
**Pull request by ramonza**

### CODE REVIEW

File: plugin/http/httptrace/httptrace_test.go [hunk 1]
```diff
@@ -23,6 +23,14 @@ import (
 	"testing"
 
 	"go.opencensus.io/trace"
+	"go.opencensus.io/trace/propagation"
+	"strings"
```
Comment (odeke-em) [inline plugin/http/httptrace/httptrace_test.go]: Minor nit, could we move the stdlib imports to make them part of the imports in line 23 and keep the blank line for the ones for `"go.opencensus.io/*"` in their own small block.

File: plugin/http/httptrace/httptrace_test.go [hunk 2]
```diff
@@ -147,3 +155,126 @@ func TestHandler(t *testing.T) {
 		})
 	}
 }
+
+var _ http.RoundTripper = &Transport{}
```
Comment (odeke-em) [inline plugin/http/httptrace/httptrace_test.go]: Minor nit:
```go
var _ http.RoundTripper = (*Transport)(nil)
```

File: plugin/http/httptrace/httptrace_test.go [hunk 3]
```diff
@@ -147,3 +155,126 @@ func TestHandler(t *testing.T) {
...
+		} else if strings.HasPrefix(sp.Name, "Recv.") {
+			server = sp
+			if server.Name != "Recv./example/url/path" {
+				t.Errorf("invalid name: %s", server.Name)
+			}
+		}
+	}
+
+	if server == nil || client == nil {
+		t.Errorf("server or client span missing")
```
Comment (odeke-em) [inline plugin/http/httptrace/httptrace_test.go]: t.Fatalf
and then we can get rid of the else clause

File: plugin/http/httptrace/httptrace_test.go [hunk 4]
```diff
@@ -147,3 +155,126 @@ func TestHandler(t *testing.T) {
...
+	} else {
+		if server.TraceID != client.TraceID {
+			t.Errorf("TraceID does not match: %s vs %s", server.TraceID, client.TraceID)
```
Comment (odeke-em) [inline plugin/http/httptrace/httptrace_test.go]:
```go
t.Errorf("TraceID does not match: serverTraceID=%q clientTraceID=%q", server.TraceID, client.TraceID)
```

**Comment by odeke-em** [comment]
Thank you @ramonza for finding and reporting this bug and also for the test. I've made PR https://github.com/census-instrumentation/opencensus-go/pull/329, whose test is needed for this.

**Comment by ramonza** [comment]
@odeke-em updated and skipped pending fix

**Comment by odeke-em** [comment]
Nice, am enjoying this! Aye aye, would you like me to first update the fix and then we can merge yours?

File: plugin/http/httptrace/httptrace_test.go [hunk 5]
```diff
@@ -147,3 +156,120 @@ func TestHandler(t *testing.T) {
...
+		t.Fatalf("unexpected stats: %d", resp.StatusCode)
+	}
+
+	serverReturn <- time.Now().Add(time.Millisecond)
+
+	respBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		t.Fatalf("unexpected read error: %#v", err)
+	}
+	if string(respBody) != "expected-response" {
```
Comment (rakyll) [inline plugin/http/httptrace/httptrace_test.go]: got := string(respBody); got != "..." {
t.Fatalf("... response: %s", got)
}

**Pull request merged**